### DISCUSSION TITLE
Docker image with support for X11 display forwarding

### DISCUSSION THREAD

**Discussion by bearddan2000** [aardappel/lobster#258] [Show and tell]
First thank you for all your efforts on this project. I enjoy finding and using new languages.

I would like to purpose a docker / podman image for lobster. Ideally, this should be an alpine image but I could not find a replacement for mesa-common-dev. These two images are ubuntu based, unfortuitously.

There are two files because there is a make and ninja versions. Probably based on how long one wants tthe build to take.

Lastly, both files / images support X11 display forwarding. X11 display forwarding allows develop of GUI game wiindows running inside a docker container. The advantage of X11 display forwarding in docker is that you can now use other docker images like databases and caches.

Examples assume:
- running ubuntu 20.40 or greater
- docker is installed
 
running with cli with make version:
&& docker build -t lobster-example Dockerfile-make \
&& docker run --rm lobster-example

running with cli with ninja version:
docker build -t lobster-example Dockerfile-ninja \
&& docker run --rm lobster-example

running with X11 display forwarding with make version:
xhost + local:docker \
&& docker build -t lobster-example Dockerfile-make \
&& docker run --rm -v /tmp/.X11-unix:/tmp/.X11-unix lobster-example

running with X11 display forwarding with ninja version:
xhost + local:docker \
&& docker build -t lobster-example Dockerfile-ninja \
&& docker run --rm -v /tmp/.X11-unix:/tmp/.X11-unix lobster-example

One suggestion, check your install docs, I found some problems which I corrected in these docker files.

[lobster-docker.tar.gz](https://github.com/aardappel/lobster/files/12466090/lobster-docker.tar.gz)

**Comment by aardappel** [comment]
Thanks for trying this out! I am trying to keep things simple / fast in terms of updating Lobster though, so building docker images, especially ones so specific, doesn't sound like something I'd want to be involved in.
